ui/apps/platform/eslint-plugins/pluginGeneric.js

```diff
@@ -75,6 +75,162 @@ const rules = {
+                                    );
+                                if (!defaultColumnProperty) {
+                                    context.report({
+                                        node,
+                                        message: `Td has className={getVisibilityClass("${argument}")} but argument is not a key in defaultColumns`,
+                                    });
+                                } else {
+                                    const dataLabelAttribute = node.attributes.find(
+                                        (attribute) =>
+                                            attribute.name?.name === 'dataLabel' &&
+                                            typeof attribute.value?.value === 'string'
+                                    );
+                                    // Another rule reports absence of dataLabel prop in Td element.
+                                    if (dataLabelAttribute) {
+                                        const dataLabel = dataLabelAttribute.value.value;
+                                        const title = defaultColumnProperty.value?.properties?.find(
+                                            (property) => property.key?.name === 'title'
+                                        )?.value?.value;
+                                        // TypeScript reports absence of title property in default column property.
+                                        if (dataLabel !== title) {
+                                            // Another rule reports inconsistency between Td dataLabel and Th text.
+                                            context.report({
+                                                node,
+                                                message: `Td has dataLabel="${dataLabel}" but defaultColumns has ${argument}: {title: "${title}"}`,
+                                            });
+                                        }
+                                    }
+                                }
+                            }
+                        }
+                    }
+                },
+            };
+        },
+    },
+    'Th-defaultColumns': {
+        // Require that Th element has key and text from defaultColumns configuration.
+        // That is, if Th element has prop for column management:
+        // className={getVisibilityClass('whichever')}
+        // Whatever as text
+        // Then defaultColumns has whichever: {title: "Whatever"} property.
+        meta: {
+            type: 'problem',
+            docs: {
+                description:
+                    'Require that Th element has key and text from defaultColumns configuration',
+            },
+            schema: [],
+        },
+        create(context) {
+            return {
+                JSXOpeningElement(node) {
+                    if (typeof node.name?.name === 'string' && node.name.name.endsWith('Th')) {
```

Comment:
That is a super question.

I feel like this ended up double minded, with both positive and negative logic.
A more consistent pattern seems easier to read, especially because lint rules use imperative code to achieve a declarative goal.